### PR TITLE
Add enterprise search to the stack

### PR DESCRIPTION
**Pull request by fsmolinski-lp**
- adding Enterprise Search (App Search + Workplace Search) to the stack
- switching Kibana service to plain HTTP, because Enterprise Search does not support custom certs when connecting to Kibana
- adding dedicated metricbeat for Enterprise Search monitoring (to show it in cluster monitoring tab)

### CODE REVIEW

File: charts/elastic/1.3.0/templates/enterprise-search.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    enterprisesearch.k8s.elastic.co/name: {{ include "common.names.shortname" $ }}
+
+spec:
+  rules:
+  - host: {{ .Values.enterprise_search.params.ingress.hostname }}
+    http:
+      paths:
+      - backend:
+          service:
+            name: {{ include "common.names.shortname" $ -}}-enterprise-search-ent-http
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/enterprise-search.yaml:64]:
```suggestion
            name: {{ include "common.names.shortname" $ -}}-entsearch-ent-http
```

File: charts/elastic/1.3.0/templates/kibana.yaml [hunk 2]
```diff
@@ -81,7 +81,7 @@ spec:
     tls:
       {{- if not $.Values.tlsSecretName }}
       selfSignedCertificate:
-        disabled: false
+        disabled: true
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/kibana.yaml:84]:
```suggestion
        disabled: false
```

File: charts/elastic/1.3.0/templates/kibana.yaml [hunk 3]
```diff
@@ -81,7 +81,7 @@ spec:
...
         subjectAltNames:
         - dns: {{ .ingress.hostname | quote }}
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/kibana.yaml:86]:
```suggestion
        - dns: {{ .ingress.hostname | quote }}
        {{- if and $.Values.enterprise_search.enabled  $.Values.enterprise_search.params.ingress.hostname }}
        - dns: "{{ include "common.names.shortname" $ -}}-entsearch-ent-http"
        - dns: {{ $.Values.enterprise_search.params.ingress.hostname }}
        {{- end }}
```

File: charts/elastic/1.3.0/values.yaml [hunk 4]
```diff
@@ -329,6 +329,36 @@ logstash:
     tolerations: []
     pipelines: []
 
+enterprise_search:
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/values.yaml:332]:
```suggestion
enterpriseSearch:
```
Please change it in all values

File: charts/elastic/1.3.0/values.yaml [hunk 5]
```diff
@@ -329,6 +329,36 @@ logstash:
...
+  params:
+    count: 1
+    image:
+      registry: docker.elastic.co
+      repository: package-registry/distribution
+      tag: ""
+    cpu: "2000m"
+    memory: "8Gi"
+    xms: "7500m"
+    xmx: "7500m"
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/values.yaml:343]:
```suggestion

```

File: charts/elastic/1.3.0/templates/kibana.yaml [hunk 6]
```diff
@@ -94,6 +94,8 @@ spec:
       {{- end }}
   elasticsearchRef:
     name: {{ include "common.names.shortname" $ }}
+  enterpriseSearchRef:
+    name: {{ include "common.names.shortname" $ }}-enterprise-search
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/kibana.yaml:98]:
```suggestion
    name: {{ include "common.names.shortname" $ }}-entsearch
```

File: charts/elastic/1.3.0/templates/enterprise-search.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  count: {{ .Values.enterprise_search.params.count }}
+  elasticsearchRef:
+    name: {{ include "common.names.shortname" $ }}
+  podTemplate:
+    metadata:
+      labels:
+        scrape: entsearch-{{ include "common.names.namespace" $ }}
+    spec:
+      containers:
+      - name: enterprise-search
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/enterprise-search.yaml:21]:
```suggestion
      - name: enterprise-search
        volumeMounts: 
        - mountPath: /usr/share/kibana/config/kibana-certs
          name: kibana-certs
          readOnly: true

```

File: charts/elastic/1.3.0/templates/enterprise-search.yaml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            cpu: {{ .Values.enterprise_search.params.cpu }}
+            memory: {{ .Values.enterprise_search.params.memory }}
+        env:
+          - name: JAVA_OPTS
+            value: -Xms{{ .Values.enterprise_search.params.xms }} -Xmx{{ .Values.enterprise_search.params.xmx }}
+  config:
+    # define the exposed URL at which users will reach Enterprise Search
+    ent_search.external_url: https://{{ .Values.enterprise_search.params.ingress.hostname }}
+    allow_es_settings_modification: true
+    kibana.external_url: https://{{ .Values.kibana.params.ingress.hostname }}
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/enterprise-search.yaml:33]:
```suggestion
    {{- if $.Values.enterprise_search.params.ingress.enabled }}
    kibana.external_url: https://{{ .Values.kibana.params.ingress.hostname }}
    {{- else }}
    kibana.external_url: https://{{ include "common.names.shortname" $ }}-kb-http.{{ include "common.names.namespace" . }}.svc:5601
    {{- end }}
```

File: charts/elastic/1.3.0/templates/enterprise-search.yaml [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    # kibana.host no tls because of https://www.elastic.co/guide/en/enterprise-search/current/configure-ssl-tls.html#configure-ssl-tls-custom-ca-server-requests
+    kibana.host: http://{{ include "common.names.shortname" $ }}-kb-http.{{ include "common.names.namespace" . }}.svc:5601
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/enterprise-search.yaml:35]:
```suggestion
    kibana.host: http://{{ include "common.names.shortname" $ }}-kb-http.{{ include "common.names.namespace" . }}.svc:5601
```

File: charts/elastic/1.3.0/templates/enterprise-search.yaml [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        resources:
+          requests:
+            cpu: {{ .Values.enterprise_search.params.cpu }}
+            memory: {{ .Values.enterprise_search.params.memory }}
+        env:
+          - name: JAVA_OPTS
+            value: -Xms{{ .Values.enterprise_search.params.xms }} -Xmx{{ .Values.enterprise_search.params.xmx }}
+  config:
+    # define the exposed URL at which users will reach Enterprise Search
+    ent_search.external_url: https://{{ .Values.enterprise_search.params.ingress.hostname }}
```
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/enterprise-search.yaml:31]:
```suggestion
    ent_search.external_url: https://{{ .Values.enterprise_search.params.ingress.hostname }}
    enterpriseSearch.ssl.verificationMode: none
```
https://www.elastic.co/guide/en/enterprise-search/current/configure-ssl-tls.html#configure-ssl-tls-in-kibana

File: charts/elastic/1.3.0/templates/kibana.yaml
Comment (qwiatu-lp) [inline charts/elastic/1.3.0/templates/kibana.yaml:1]: add to kibana initContainer which checking avability of enterprise-search

File: charts/elastic/1.3.0/templates/enterprise-search.yaml (hunk 9 above)
Comment (fsmolinski-lp) [reply charts/elastic/1.3.0/templates/enterprise-search.yaml:35]: there is no change in the code here.

Comment (qwiatu-lp) [reply charts/elastic/1.3.0/templates/enterprise-search.yaml:35]: https://www.elastic.co/guide/en/enterprise-search/current/configure-ssl-tls.html#configure-ssl-tls-in-kibana

File: charts/elastic/1.3.0/templates/enterprise-search.yaml (hunk 7 above)
Comment (qwiatu-lp) [reply charts/elastic/1.3.0/templates/enterprise-search.yaml:21]: you need to add volumes in PodTemplate.spec point to kibana secret

File: charts/elastic/1.3.0/templates/kibana.yaml
Comment (fsmolinski-lp) [reply charts/elastic/1.3.0/templates/kibana.yaml:1]: cant do. it is funny that kibana must be up for https://{{ include "common.names.shortname" $ -}}-entsearch-ent-http.{{ include "common.names.namespace" $ -}}.svc:3002 to respond.

**Review by qwiatu-lp — COMMENTED** [review]
added review

**Review by qwiatu-lp — APPROVED** [review]
released 1.3.1 and 1.4.0

**Pull request merged**